transform/connectors/mp_pc.py

```diff
@@ -0,0 +1,76 @@
+import os
+import requests
+from logging import getLogger
+from dotenv import load_dotenv
+
+from schemas.properties.env import (
+    SERVICE,
+    DATASOURCE,
+    PROVIDER,
+    OFFER,
+    BUNDLE,
+)
+
+logger = getLogger(__name__)
+
+load_dotenv()
+
+
+def get_data(data_type: str, data_address: str) -> list[dict] | None:
+    try:
+        if data_type in (SERVICE, DATASOURCE, PROVIDER, OFFER, BUNDLE):
+            headers = {
+                "accept": "application/json",
+                "X-User-Token": os.environ.get("MP_API_TOKEN", default=""),
+            }
+            data = requests.get(data_address, headers=headers, timeout=1000).json()
+
+        else:
+            data = requests.get(data_address, timeout=20).json()["results"]
+
+    except requests.ConnectionError:
+        data = None
+        logger.error(
+            f"{data_type} - retrieving data from {data_address} failed. Unable to establish a connection to the server"
+        )
+    except requests.Timeout:
+        data = None
+        logger.error(
+            f"{data_type} - retrieving data from {data_address} failed. The request timed out"
+        )
+    except requests.HTTPError as http_err:
+        data = None
+        logger.error(
+            f"{data_type} - retrieving data from {data_address} failed. HTTP error occurred: {http_err}"
+        )
+    except requests.RequestException as err:
+        data = None
+        logger.error(
+            f"{data_type} - retrieving data from {data_address} failed. An error occurred: {err}"
+        )
+
+    return data
+
+
+DATA_SOURCE_ADDRESS = os.environ.get("MP_API_ADDRESS") + "/api/v1/ess/datasources"
```

Comment:
it can be retrieved from env_vars right? `env_vars[ALL_COLLECTION][DATA_SOURCE][ADDRESS]` I think? Then you dont need to load env vars here

Reply:
This was the only way to do it without changing loader.py. But I managed to add to loader.py the same properties that exist in the development branch. Now it gets them from env_vars.